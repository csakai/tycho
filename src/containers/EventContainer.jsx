import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import * as EventActions from '../actions/EventActions';
import ReduxService from '../services/ReduxService';

export class EventContainer extends React.Component {

  static propTypes = {
      action: PropTypes.object,
      touched: PropTypes.number,
      released: PropTypes.number,
      onWheel: PropTypes.func
  }

  /**
   * Emits the timestamp of when container was last touched.
   */
  onTouched = () => {
      this.props.action.setTouched(Date.now());
  }

  /**
   * Emits the timestamp of when container was last released from touch.
   */
  onReleased = () => {
      this.props.action.setReleased(Date.now());
  }

  render() {
      return (
          <div
              onWheel={this.props.onWheel}
              onTouchStart={this.onTouched}
              onMouseDown={this.onTouched}>
              {this.props.children}
          </div>
      );
  }
}

export default connect(
    ReduxService.mapStateToProps(
        'event.touched',
        'event.released'
    ),
    ReduxService.mapDispatchToProps(
        EventActions
    )
)(EventContainer);