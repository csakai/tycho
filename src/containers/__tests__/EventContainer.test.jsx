import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {EventContainer} from '../EventContainer';

describe('Event Container', () => {
  let eventContainer, component;

  beforeEach(() => {
    component = shallow(<EventContainer />);
    eventContainer = component.instance();
  });

  describe('Time-broadcasting Events', () => {
    const now = Date.now();

    beforeEach(() => {
      Date.now = () => now;
      eventContainer.props = {
        action: {
          setTouched: jest.fn(),
          setReleased: jest.fn()
        }
      }
    });

    describe('onTouched()', () => {
      it('should call the global setTouched() action with current timestamp', () => {
        const spy = jest.spyOn(eventContainer.props.action, 'setTouched');

        eventContainer.onTouched();

        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith(now);
      });
    });

    describe('onReleased()', () => {
      it('should call the global setReleased() action with current timestamp', () => {
        const spy = jest.spyOn(eventContainer.props.action, 'setReleased');

        eventContainer.onReleased();

        expect(spy).toHaveBeenCalled();
        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith(now);
      });
    });
  });

  describe('render()', () => {
    it('should render the events container succesfully', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});