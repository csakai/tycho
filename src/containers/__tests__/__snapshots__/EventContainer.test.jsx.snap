// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Container render() should render the events container succesfully 1`] = `
<div
  onMouseDown={[Function]}
  onTouchStart={[Function]}
/>
`;
